utils/flink-components-testkit/src/main/scala/pl/touk/nussknacker/engine/flink/util/test/FlinkTestScenarioRunner.scala

```diff
@@ -46,10 +47,22 @@ class FlinkTestScenarioRunner(val components: List[ComponentDefinition], val con
 
     //todo get flink mini cluster through composition
     val env = flinkMiniCluster.createExecutionEnvironment()
-    val registrar = FlinkProcessRegistrar(new FlinkProcessCompilerWithTestComponents(testComponentHolder, modelData), ExecutionConfigPreparer.unOptimizedChain(modelData))
-    registrar.register(new StreamExecutionEnvironment(env), scenario, ProcessVersion.empty, DeploymentData.empty, Some(testComponentHolder.runId))
-    env.executeAndWaitForFinished(scenario.id)()
-    testComponentHolder.results(testComponentHolder.runId).map((k: Any) => k.asInstanceOf[R])
+
+    //It's copied from registrar.register only for handling compilation errors..
+    //TODO: figure how to get compilation result on highest level - registrar.register?
+    val compiler = new FlinkProcessCompilerWithTestComponents(testComponentHolder, modelData)
```

Comment:
I'd rather do it a bit differently.
Let's do sth like:
```
modelData.validator.validate(scenario).andThen { _ =>
   ... proceed with former results
}
```
We won't have to change FlinkProcessCompiler, and we'll mimic 'normal' designer behaviour: first validation, then compilation, assuming that scenario validates

Reply:
Nice! I didn't know we can do it in that way - thanks! :)